lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -224,6 +231,34 @@ def rescuable_exceptions
       def rescue_from
         []
       end
+
+      def extended_cloned_config(tenant)
+        @config.clone.with_indifferent_access.tap do |config|
+          config.merge!(Apartment.db_config_for(tenant))
```

Comment:
its convenient to merge because:
- whenever you need to work on another server, it spares a few keys in the specific config (basically adapter, pooling conf etc...)
- whenever you want to keep a tenant on the same server, you just pass an empty hash

@bradrobertson whats your preference?